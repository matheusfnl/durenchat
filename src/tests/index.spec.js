import { describe, it, expect, vi, beforeAll } from 'vitest';
import { durenchat } from '../index.ts';
import 'emoji-picker-element';

const fixedDate = new Date('2024-12-12T00:00:00Z');

vi.useFakeTimers();
vi.setSystemTime(fixedDate);

describe('Test Durenchat class', () => {
  const chat = durenchat({
    users: [{
      id: 1,
      name: 'Matheus',
      color: '#4a3d1f',
      textColor: '#fff',
      photoUrl: '',
    }],
    self: 1,
  });

  it('should get user', () => {
    expect(chat.getUser(1)).toEqual({
      id: 1,
      name: 'Matheus',
      color: '#4a3d1f',
      textColor: '#fff',
      photoUrl: '',
    });
  });

  it('should not get user', () => {
    expect(() => chat.getUser(2)).toThrowError('User not found');
  });

  it('should define header', () => {
    chat.defineHeader({
      photoUrl: '',
      name: 'Matheus',
    });

    expect(chat.wrapperElement).toMatchSnapshot()
  });

  it('should define chat container', () => {
    chat.defineChatcontainer('dc-messages');

    expect(chat.wrapperElement).toMatchSnapshot()
  });

  it('should define footer', () => {
    chat.defineFooter();

    expect(chat.wrapperElement).toMatchSnapshot()
  });

  it('should send text', () => {
    const message = chat.sendMessage({
      sender: 1,
      content: 'Olá',
    });

    expect(message.content).toEqual('Olá');
    expect(message.status).toEqual('sent');
    expect(message.sender).toEqual({
      id: 1,
      name: 'Matheus',
      color: '#4a3d1f',
      textColor: '#fff',
      photoUrl: '',
    });
  });

  it('should send image', () => {
    const message = chat.sendMessage({
      sender: 1,
      content: {
        type: 'image',
        url: 'https://example.com/image.jpg',
        caption: 'Texto',
      },
    });

    expect(message.content).toEqual({
      type: 'image',
      url: 'https://example.com/image.jpg',
      caption: 'Texto',
    });

    expect(message.status).toEqual('sent');
    expect(message.sender).toEqual({
      id: 1,
      name: 'Matheus',
      color: '#4a3d1f',
      textColor: '#fff',
      photoUrl: '',
    });
  });

  it('should send audio', () => {
    const message = chat.sendMessage({
      sender: 1,
      content: {
        type: 'audio',
        url: 'https://example.com/audio.mp3',
      },
    });

    expect(message.content).toEqual({
      type: 'audio',
      url: 'https://example.com/audio.mp3',
    });

    expect(message.status).toEqual('sent');
    expect(message.sender).toEqual({
      id: 1,
      name: 'Matheus',
      color: '#4a3d1f',
      textColor: '#fff',
      photoUrl: '',
    });
  });

  it('should send video', () => {
    const message = chat.sendMessage({
      sender: 1,
      content: {
        type: 'video',
        url: 'https://example.com/video.mp4',
      },
    });

    expect(message.content).toEqual({
      type: 'video',
      url: 'https://example.com/video.mp4',
    });

    expect(message.status).toEqual('sent');
    expect(message.sender).toEqual({
      id: 1,
      name: 'Matheus',
      color: '#4a3d1f',
      textColor: '#fff',
      photoUrl: '',
    });
  });

  it('should send document', () => {
    const message = chat.sendMessage({
      sender: 1,
      content: {
        type: 'document',
        url: 'https://example.com/document.pdf',
      },
    });

    expect(message.content).toEqual({
      type: 'document',
      url: 'https://example.com/document.pdf',
    });

    expect(message.status).toEqual('sent');
    expect(message.sender).toEqual({
      id: 1,
      name: 'Matheus',
      color: '#4a3d1f',
      textColor: '#fff',
      photoUrl: '',
    });
  });

  it('should match snapshot', () => {
    expect(chat).toMatchSnapshot();
  })
})