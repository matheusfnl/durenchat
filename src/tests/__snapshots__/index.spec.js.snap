// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test Durenchat class > should define chat container 1`] = `
<div
  style="height: 100%; display: flex; flex-direction: column; box-sizing: border-box;"
>
  <div
    class="header-chat"
  >
    <img
      alt="Matheus photo"
      class="img-header-chat"
      src=""
    />
    <div
      class="name-header-chat"
    >
      Matheus
    </div>
  </div>
  <div
    class="chat-container"
    id="dc-messages"
  />
</div>
`;

exports[`Test Durenchat class > should define footer 1`] = `
<div
  style="height: 100%; display: flex; flex-direction: column; box-sizing: border-box;"
>
  <div
    class="header-chat"
  >
    <img
      alt="Matheus photo"
      class="img-header-chat"
      src=""
    />
    <div
      class="name-header-chat"
    >
      Matheus
    </div>
  </div>
  <div
    class="chat-container"
    id="dc-messages"
  />
  <div
    class="footer-chat"
  >
    <div
      class="footer-container"
    />
    <div
      class="footer-container"
    />
    <input
      class="input-text"
      placeholder="Digite uma mensagem..."
      type="text"
    />
    <div
      class="footer-container"
    />
    <input
      accept="image/*,video/*,application/pdf,audio/mpeg"
      style="display: none;"
      type="file"
    />
  </div>
</div>
`;

exports[`Test Durenchat class > should define header 1`] = `
<div
  style="height: 100%; display: flex; flex-direction: column; box-sizing: border-box;"
>
  <div
    class="header-chat"
  >
    <img
      alt="Matheus photo"
      class="img-header-chat"
      src=""
    />
    <div
      class="name-header-chat"
    >
      Matheus
    </div>
  </div>
</div>
`;

exports[`Test Durenchat class > should match snapshot 1`] = `
Durenchat {
  "audioFooterElement": null,
  "chatElement": <div
    class="chat-container"
    id="dc-messages"
  >
    <div
      class="message-container message-container-sender"
    >
      <div
        class="chat-message"
        style="background-color: rgb(74, 61, 31); color: rgb(255, 255, 255); position: relative;"
      >
        <div
          class="chat-message-content"
        >
          Olá
        </div>
        <div
          class="chat-message-info"
        >
          <div
            class="chat-message-date"
            style="opacity: 0.7;"
          >
            21:00
          </div>
          <div
            class="chat-message-status"
          />
        </div>
        <span
          data-message-id="1"
          style="position: absolute; top: 6px; right: 6px; background-color: rgb(74, 61, 31); display: none; cursor: pointer; border-radius: 4px; box-shadow: 0 0 5px 5px #4a3d1f80;"
        >
          <div
            class="options-icon"
          />
        </span>
      </div>
    </div>
    <div
      class="message-container message-container-sender"
    >
      <div
        class="chat-message"
        style="background-color: rgb(74, 61, 31); color: rgb(255, 255, 255); position: relative;"
      >
        <div
          class="chat-message-content"
        >
          <img
            alt="Imagem"
            class="max-message-content"
            src="https://example.com/image.jpg"
          />
          <div
            class="message-caption"
          >
            Texto
          </div>
        </div>
        <div
          class="chat-message-info"
        >
          <div
            class="chat-message-date"
            style="opacity: 0.7;"
          >
            21:00
          </div>
          <div
            class="chat-message-status"
          />
        </div>
        <span
          data-message-id="2"
          style="position: absolute; top: 6px; right: 6px; background-color: rgb(74, 61, 31); display: none; cursor: pointer; border-radius: 4px; box-shadow: 0 0 5px 5px #4a3d1f80;"
        >
          <div
            class="options-icon"
          />
        </span>
      </div>
    </div>
    <div
      class="message-container message-container-sender"
    >
      <div
        class="chat-message"
        style="background-color: rgb(74, 61, 31); color: rgb(255, 255, 255); position: relative;"
      >
        <div
          class="chat-message-content"
        >
          <audio
            class="max-message-content"
            controls=""
          >
            <source
              src="https://example.com/audio.mp3"
              type="audio/mpeg"
            />
          </audio>
        </div>
        <div
          class="chat-message-info"
        >
          <div
            class="chat-message-date"
            style="opacity: 0.7;"
          >
            21:00
          </div>
          <div
            class="chat-message-status"
          />
        </div>
        <span
          data-message-id="3"
          style="position: absolute; top: 6px; right: 6px; background-color: rgb(74, 61, 31); display: none; cursor: pointer; border-radius: 4px; box-shadow: 0 0 5px 5px #4a3d1f80;"
        >
          <div
            class="options-icon"
          />
        </span>
      </div>
    </div>
    <div
      class="message-container message-container-sender"
    >
      <div
        class="chat-message"
        style="background-color: rgb(74, 61, 31); color: rgb(255, 255, 255); position: relative;"
      >
        <div
          class="chat-message-content"
        >
          <video
            class="max-message-content"
            controls=""
          >
            <source
              src="https://example.com/video.mp4"
              type="video/mp4"
            />
          </video>
          <div
            class="message-caption"
          />
        </div>
        <div
          class="chat-message-info"
        >
          <div
            class="chat-message-date"
            style="opacity: 0.7;"
          >
            21:00
          </div>
          <div
            class="chat-message-status"
          />
        </div>
        <span
          data-message-id="4"
          style="position: absolute; top: 6px; right: 6px; background-color: rgb(74, 61, 31); display: none; cursor: pointer; border-radius: 4px; box-shadow: 0 0 5px 5px #4a3d1f80;"
        >
          <div
            class="options-icon"
          />
        </span>
      </div>
    </div>
    <div
      class="message-container message-container-sender"
    >
      <div
        class="chat-message"
        style="background-color: rgb(74, 61, 31); color: rgb(255, 255, 255); position: relative;"
      >
        <div
          class="chat-message-content"
        >
          <a
            class="max-message-content"
            href="https://example.com/document.pdf"
            target="_blank"
          />
          <div
            class="message-caption"
          />
        </div>
        <div
          class="chat-message-info"
        >
          <div
            class="chat-message-date"
            style="opacity: 0.7;"
          >
            21:00
          </div>
          <div
            class="chat-message-status"
          />
        </div>
        <span
          data-message-id="5"
          style="position: absolute; top: 6px; right: 6px; background-color: rgb(74, 61, 31); display: none; cursor: pointer; border-radius: 4px; box-shadow: 0 0 5px 5px #4a3d1f80;"
        >
          <div
            class="options-icon"
          />
        </span>
      </div>
    </div>
  </div>,
  "contextMenuMessage": null,
  "editingMessage": null,
  "events": {},
  "footerElement": <div
    class="footer-chat"
  >
    <div
      class="footer-container"
    />
    <div
      class="footer-container"
    />
    <input
      class="input-text"
      placeholder="Digite uma mensagem..."
      type="text"
    />
    <div
      class="footer-container"
    />
    <input
      accept="image/*,video/*,application/pdf,audio/mpeg"
      style="display: none;"
      type="file"
    />
  </div>,
  "icons": {},
  "isRecording": false,
  "messages": [
    Message {
      "content": "Olá",
      "edited_at": null,
      "element": <div
        class="message-container message-container-sender"
      >
        <div
          class="chat-message"
          style="background-color: rgb(74, 61, 31); color: rgb(255, 255, 255); position: relative;"
        >
          <div
            class="chat-message-content"
          >
            Olá
          </div>
          <div
            class="chat-message-info"
          >
            <div
              class="chat-message-date"
              style="opacity: 0.7;"
            >
              21:00
            </div>
            <div
              class="chat-message-status"
            />
          </div>
          <span
            data-message-id="1"
            style="position: absolute; top: 6px; right: 6px; background-color: rgb(74, 61, 31); display: none; cursor: pointer; border-radius: 4px; box-shadow: 0 0 5px 5px #4a3d1f80;"
          >
            <div
              class="options-icon"
            />
          </span>
        </div>
      </div>,
      "id": 1,
      "sender": User {
        "color": "#4a3d1f",
        "id": 1,
        "name": "Matheus",
        "photoUrl": "",
        "textColor": "#fff",
      },
      "sent_at": 2024-12-12T00:00:00.000Z,
      "status": "sent",
    },
    Message {
      "content": {
        "caption": "Texto",
        "type": "image",
        "url": "https://example.com/image.jpg",
      },
      "edited_at": null,
      "element": <div
        class="message-container message-container-sender"
      >
        <div
          class="chat-message"
          style="background-color: rgb(74, 61, 31); color: rgb(255, 255, 255); position: relative;"
        >
          <div
            class="chat-message-content"
          >
            <img
              alt="Imagem"
              class="max-message-content"
              src="https://example.com/image.jpg"
            />
            <div
              class="message-caption"
            >
              Texto
            </div>
          </div>
          <div
            class="chat-message-info"
          >
            <div
              class="chat-message-date"
              style="opacity: 0.7;"
            >
              21:00
            </div>
            <div
              class="chat-message-status"
            />
          </div>
          <span
            data-message-id="2"
            style="position: absolute; top: 6px; right: 6px; background-color: rgb(74, 61, 31); display: none; cursor: pointer; border-radius: 4px; box-shadow: 0 0 5px 5px #4a3d1f80;"
          >
            <div
              class="options-icon"
            />
          </span>
        </div>
      </div>,
      "id": 2,
      "sender": User {
        "color": "#4a3d1f",
        "id": 1,
        "name": "Matheus",
        "photoUrl": "",
        "textColor": "#fff",
      },
      "sent_at": 2024-12-12T00:00:00.000Z,
      "status": "sent",
    },
    Message {
      "content": {
        "type": "audio",
        "url": "https://example.com/audio.mp3",
      },
      "edited_at": null,
      "element": <div
        class="message-container message-container-sender"
      >
        <div
          class="chat-message"
          style="background-color: rgb(74, 61, 31); color: rgb(255, 255, 255); position: relative;"
        >
          <div
            class="chat-message-content"
          >
            <audio
              class="max-message-content"
              controls=""
            >
              <source
                src="https://example.com/audio.mp3"
                type="audio/mpeg"
              />
            </audio>
          </div>
          <div
            class="chat-message-info"
          >
            <div
              class="chat-message-date"
              style="opacity: 0.7;"
            >
              21:00
            </div>
            <div
              class="chat-message-status"
            />
          </div>
          <span
            data-message-id="3"
            style="position: absolute; top: 6px; right: 6px; background-color: rgb(74, 61, 31); display: none; cursor: pointer; border-radius: 4px; box-shadow: 0 0 5px 5px #4a3d1f80;"
          >
            <div
              class="options-icon"
            />
          </span>
        </div>
      </div>,
      "id": 3,
      "sender": User {
        "color": "#4a3d1f",
        "id": 1,
        "name": "Matheus",
        "photoUrl": "",
        "textColor": "#fff",
      },
      "sent_at": 2024-12-12T00:00:00.000Z,
      "status": "sent",
    },
    Message {
      "content": {
        "type": "video",
        "url": "https://example.com/video.mp4",
      },
      "edited_at": null,
      "element": <div
        class="message-container message-container-sender"
      >
        <div
          class="chat-message"
          style="background-color: rgb(74, 61, 31); color: rgb(255, 255, 255); position: relative;"
        >
          <div
            class="chat-message-content"
          >
            <video
              class="max-message-content"
              controls=""
            >
              <source
                src="https://example.com/video.mp4"
                type="video/mp4"
              />
            </video>
            <div
              class="message-caption"
            />
          </div>
          <div
            class="chat-message-info"
          >
            <div
              class="chat-message-date"
              style="opacity: 0.7;"
            >
              21:00
            </div>
            <div
              class="chat-message-status"
            />
          </div>
          <span
            data-message-id="4"
            style="position: absolute; top: 6px; right: 6px; background-color: rgb(74, 61, 31); display: none; cursor: pointer; border-radius: 4px; box-shadow: 0 0 5px 5px #4a3d1f80;"
          >
            <div
              class="options-icon"
            />
          </span>
        </div>
      </div>,
      "id": 4,
      "sender": User {
        "color": "#4a3d1f",
        "id": 1,
        "name": "Matheus",
        "photoUrl": "",
        "textColor": "#fff",
      },
      "sent_at": 2024-12-12T00:00:00.000Z,
      "status": "sent",
    },
    Message {
      "content": {
        "type": "document",
        "url": "https://example.com/document.pdf",
      },
      "edited_at": null,
      "element": <div
        class="message-container message-container-sender"
      >
        <div
          class="chat-message"
          style="background-color: rgb(74, 61, 31); color: rgb(255, 255, 255); position: relative;"
        >
          <div
            class="chat-message-content"
          >
            <a
              class="max-message-content"
              href="https://example.com/document.pdf"
              target="_blank"
            />
            <div
              class="message-caption"
            />
          </div>
          <div
            class="chat-message-info"
          >
            <div
              class="chat-message-date"
              style="opacity: 0.7;"
            >
              21:00
            </div>
            <div
              class="chat-message-status"
            />
          </div>
          <span
            data-message-id="5"
            style="position: absolute; top: 6px; right: 6px; background-color: rgb(74, 61, 31); display: none; cursor: pointer; border-radius: 4px; box-shadow: 0 0 5px 5px #4a3d1f80;"
          >
            <div
              class="options-icon"
            />
          </span>
        </div>
      </div>,
      "id": 5,
      "sender": User {
        "color": "#4a3d1f",
        "id": 1,
        "name": "Matheus",
        "photoUrl": "",
        "textColor": "#fff",
      },
      "sent_at": 2024-12-12T00:00:00.000Z,
      "status": "sent",
    },
  ],
  "optionsElement": null,
  "prefix": "",
  "recorder": MicRecorder {
    "activeStream": null,
    "config": {
      "bitRate": 128,
      "deviceId": null,
      "startRecordingAt": 300,
    },
    "context": null,
    "microphone": null,
    "processor": null,
    "startTime": 0,
  },
  "self": User {
    "color": "#4a3d1f",
    "id": 1,
    "name": "Matheus",
    "photoUrl": "",
    "textColor": "#fff",
  },
  "users": [
    User {
      "color": "#4a3d1f",
      "id": 1,
      "name": "Matheus",
      "photoUrl": "",
      "textColor": "#fff",
    },
  ],
  "wrapperElement": <div
    style="height: 100%; display: flex; flex-direction: column; box-sizing: border-box;"
  >
    <div
      class="header-chat"
    >
      <img
        alt="Matheus photo"
        class="img-header-chat"
        src=""
      />
      <div
        class="name-header-chat"
      >
        Matheus
      </div>
    </div>
    <div
      class="chat-container"
      id="dc-messages"
    >
      <div
        class="message-container message-container-sender"
      >
        <div
          class="chat-message"
          style="background-color: rgb(74, 61, 31); color: rgb(255, 255, 255); position: relative;"
        >
          <div
            class="chat-message-content"
          >
            Olá
          </div>
          <div
            class="chat-message-info"
          >
            <div
              class="chat-message-date"
              style="opacity: 0.7;"
            >
              21:00
            </div>
            <div
              class="chat-message-status"
            />
          </div>
          <span
            data-message-id="1"
            style="position: absolute; top: 6px; right: 6px; background-color: rgb(74, 61, 31); display: none; cursor: pointer; border-radius: 4px; box-shadow: 0 0 5px 5px #4a3d1f80;"
          >
            <div
              class="options-icon"
            />
          </span>
        </div>
      </div>
      <div
        class="message-container message-container-sender"
      >
        <div
          class="chat-message"
          style="background-color: rgb(74, 61, 31); color: rgb(255, 255, 255); position: relative;"
        >
          <div
            class="chat-message-content"
          >
            <img
              alt="Imagem"
              class="max-message-content"
              src="https://example.com/image.jpg"
            />
            <div
              class="message-caption"
            >
              Texto
            </div>
          </div>
          <div
            class="chat-message-info"
          >
            <div
              class="chat-message-date"
              style="opacity: 0.7;"
            >
              21:00
            </div>
            <div
              class="chat-message-status"
            />
          </div>
          <span
            data-message-id="2"
            style="position: absolute; top: 6px; right: 6px; background-color: rgb(74, 61, 31); display: none; cursor: pointer; border-radius: 4px; box-shadow: 0 0 5px 5px #4a3d1f80;"
          >
            <div
              class="options-icon"
            />
          </span>
        </div>
      </div>
      <div
        class="message-container message-container-sender"
      >
        <div
          class="chat-message"
          style="background-color: rgb(74, 61, 31); color: rgb(255, 255, 255); position: relative;"
        >
          <div
            class="chat-message-content"
          >
            <audio
              class="max-message-content"
              controls=""
            >
              <source
                src="https://example.com/audio.mp3"
                type="audio/mpeg"
              />
            </audio>
          </div>
          <div
            class="chat-message-info"
          >
            <div
              class="chat-message-date"
              style="opacity: 0.7;"
            >
              21:00
            </div>
            <div
              class="chat-message-status"
            />
          </div>
          <span
            data-message-id="3"
            style="position: absolute; top: 6px; right: 6px; background-color: rgb(74, 61, 31); display: none; cursor: pointer; border-radius: 4px; box-shadow: 0 0 5px 5px #4a3d1f80;"
          >
            <div
              class="options-icon"
            />
          </span>
        </div>
      </div>
      <div
        class="message-container message-container-sender"
      >
        <div
          class="chat-message"
          style="background-color: rgb(74, 61, 31); color: rgb(255, 255, 255); position: relative;"
        >
          <div
            class="chat-message-content"
          >
            <video
              class="max-message-content"
              controls=""
            >
              <source
                src="https://example.com/video.mp4"
                type="video/mp4"
              />
            </video>
            <div
              class="message-caption"
            />
          </div>
          <div
            class="chat-message-info"
          >
            <div
              class="chat-message-date"
              style="opacity: 0.7;"
            >
              21:00
            </div>
            <div
              class="chat-message-status"
            />
          </div>
          <span
            data-message-id="4"
            style="position: absolute; top: 6px; right: 6px; background-color: rgb(74, 61, 31); display: none; cursor: pointer; border-radius: 4px; box-shadow: 0 0 5px 5px #4a3d1f80;"
          >
            <div
              class="options-icon"
            />
          </span>
        </div>
      </div>
      <div
        class="message-container message-container-sender"
      >
        <div
          class="chat-message"
          style="background-color: rgb(74, 61, 31); color: rgb(255, 255, 255); position: relative;"
        >
          <div
            class="chat-message-content"
          >
            <a
              class="max-message-content"
              href="https://example.com/document.pdf"
              target="_blank"
            />
            <div
              class="message-caption"
            />
          </div>
          <div
            class="chat-message-info"
          >
            <div
              class="chat-message-date"
              style="opacity: 0.7;"
            >
              21:00
            </div>
            <div
              class="chat-message-status"
            />
          </div>
          <span
            data-message-id="5"
            style="position: absolute; top: 6px; right: 6px; background-color: rgb(74, 61, 31); display: none; cursor: pointer; border-radius: 4px; box-shadow: 0 0 5px 5px #4a3d1f80;"
          >
            <div
              class="options-icon"
            />
          </span>
        </div>
      </div>
    </div>
    <div
      class="footer-chat"
    >
      <div
        class="footer-container"
      />
      <div
        class="footer-container"
      />
      <input
        class="input-text"
        placeholder="Digite uma mensagem..."
        type="text"
      />
      <div
        class="footer-container"
      />
      <input
        accept="image/*,video/*,application/pdf,audio/mpeg"
        style="display: none;"
        type="file"
      />
    </div>
  </div>,
}
`;
